src/plugins/better-evaluation-view/components/ListEvaluation.tsx

```diff
@@ -1,58 +1,185 @@
 import { jsx } from "#DCGView";
 import { StaticMathQuillView } from "#components";
-import { Private, TypedConstantValue, ValueType } from "#globals";
+import {
+  LabelOptionsBase,
+  Private,
+  TypedConstantValue,
+  ValueType,
+  ValueTypeMap,
+} from "#globals";
+import { autoCommands, autoOperatorNames } from "#utils/depUtils.ts";
+import { zipWith } from "#utils/utils.ts";
+
+const labelOptions = {
+  smallCutoff: 0.00001,
+  bigCutoff: 1000000,
+  digits: 5,
+  displayAsFraction: false,
+} satisfies LabelOptionsBase;
+
+const { Label } = Private.Mathtools;
+
+type ComplexNumberLabel = typeof Label.complexNumberLabel;
+type PointLabel = typeof Label.pointLabel;
+type Point3dLabel = typeof Label.point3dLabel;
+type TruncatedLatexLabel = typeof Label.truncatedLatexLabel;
+
+const uprightUndefined = (label: string) =>
+  label === "undefined" ? "\\mathrm{undefined}" : label;
+
+const complexNumberLabel = (label: Parameters<ComplexNumberLabel>[0]) =>
+  uprightUndefined(Label.complexNumberLabel(label, labelOptions));
+const pointLabel = (label: Parameters<PointLabel>[0]) =>
+  uprightUndefined(Label.pointLabel(label, labelOptions));
+const point3dLabel = (label: Parameters<Point3dLabel>[0]) =>
+  uprightUndefined(Label.point3dLabel(label, labelOptions));
+const truncatedLatexLabel = (label: Parameters<TruncatedLatexLabel>[0]) =>
+  uprightUndefined(Label.truncatedLatexLabel(label, labelOptions));
+
+type NormalListValueType =
+  | ValueType.ListOfComplex
+  | ValueType.ListOfAny
+  | ValueType.ListOfNumber
+  | ValueType.ListOfBool
+  | ValueType.ListOfPoint
+  | ValueType.ListOfPoint3D
+  | ValueType.EmptyList
+  | ValueType.ListOfPolygon
+  | ValueType.ListOfSegment
+  | ValueType.ListOfCircle
+  | ValueType.ListOfArc
+  | ValueType.ListOfLine
+  | ValueType.ListOfRay
+  | ValueType.ListOfVector
+  | ValueType.ListOfSegment3D
+  | ValueType.ListOfTriangle3D
+  | ValueType.ListOfSphere3D
+  | ValueType.ListOfVector3D
+  | ValueType.ListOfTone;
+
+type TypedConstantIteratorValue<T extends NormalListValueType> = T extends T
+  ? {
+      valueType: T;
+      iterator: ArrayIterator<ValueTypeMap[T][number]>;
+    }
+  : never;
+
+function formatLabels<T extends NormalListValueType>(
+  typedConstantValue: TypedConstantValue<T>
+): IteratorObject<string> {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+  const { valueType, iterator } = {
+    valueType: typedConstantValue.valueType,
+    iterator: typedConstantValue.value.values(),
+  } as TypedConstantIteratorValue<T>;
+  switch (valueType) {
+    case ValueType.ListOfComplex:
+      return iterator.map(complexNumberLabel);
+    case ValueType.ListOfAny:
+      return iterator.map(String);
```

Comment:
I suspect this is unreachable; if Desmos starts producing ListOfAny, this could end up giving `[Object object]` type stuff. Doubt there's a better idea though.